pkg/catalog/catalog.go

```diff
@@ -1744,38 +1742,38 @@ func (m *GCUncommittedMark) Encode() (string, error) {
 	return result, nil
 }
 
+type PrepareGCUncommittedInfo struct {
+	Mark     *GCUncommittedMark
+	Metadata *graveler.GarbageCollectionRunMetadata
+}
+
 type UncommittedParquetObject struct {
 	PhysicalAddress string `parquet:"name=physical_address, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN_DICTIONARY"`
 	CreationDate    int64  `parquet:"name=creation_date, type=INT64, convertedtype=INT_64"`
 }
 
-func (c *Catalog) writeUncommittedLocal(ctx context.Context, repository *graveler.RepositoryRecord, fd *os.File, mark *GCUncommittedMark) (*GCUncommittedMark, error) {
-	const parallelNum = 1                                                                        // Number of goroutines to handle marshaling of data
-	pw, err := writer.NewParquetWriterFromWriter(fd, new(UncommittedParquetObject), parallelNum) // TODO: Play with np count
+func (c *Catalog) writeUncommittedLocal(ctx context.Context, repository *graveler.RepositoryRecord, w UncommittedWriter, mark *GCUncommittedMark) (*GCUncommittedMark, error) {
+	const (
+		parallelNum              = 1                // Number of goroutines to handle marshaling of data
+		GCMaxUncommittedFileSize = 20 * 1024 * 1024 // 20 MB
+		// Calculation of size deviation by periodicCheckSize value
+		// "data" prefix = 4 bytes
+		// partition id = 20 bytes (xid)
+		// object name = 20 bytes (xid)
+		// timestamp (int64) = 8 bytes
+		//
+		// Total per entry ~52 bytes
+		// Deviation with periodicCheckSize = 100000 will be around 5 MB
+		periodicCheckSize = 100000
+	)
```

Comment:
- Consider moving these constants outside the func for better visibility.
- GCMaxUncommittedFileSize -> gcMaxUncommittedFileSize